code/modules/client/loadout_picker.dm

```diff
@@ -0,0 +1,87 @@
+/datum/loadout_picker/ui_static_data(mob/user)
+	. = ..()
+
+	.["categories"] = list()
+	for(var/category in GLOB.gear_datums_by_category)
+		var/list/datum/gear/gears = GLOB.gear_datums_by_category[category]
+
+		var/items = list()
+
+		for(var/gear_key as anything in gears)
+			var/datum/gear/gear = gears[gear_key]
+			items += list(
+				list("name" = gear.display_name, "cost" = gear.cost, "icon" = gear.path::icon, "icon_state" = gear.path::icon_state)
+			)
+
+		.["categories"] += list(
+			list("name" = category, "items" = items)
+		)
+
+	.["max_points"] = MAX_GEAR_COST
+
+/datum/loadout_picker/ui_data(mob/user)
+	. = ..()
+
+	var/datum/preferences/prefs = user.client.prefs
+	if(!prefs)
+		return
+
+	var/points = 0
+
+	.["loadout"] = list()
+
+	for(var/item in prefs.gear)
+		var/datum/gear/gear = GLOB.gear_datums_by_name[item]
+		points += gear.cost
+
+		.["loadout"] += list(
+			list("name" = gear.display_name, "cost" = gear.cost, "icon" = gear.path::icon, "icon_state" = gear.path::icon_state)
+		)
+
+	.["points"] = points
+
+/datum/loadout_picker/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+
+	var/datum/preferences/prefs = ui.user.client.prefs
```

Comment:
```suggestion
	var/datum/preferences/prefs = ui.user.client?.prefs
```

Also any particular reason why you null tested prefs in ui_data but not here?

Reply:
the whole tgui chain should prevent a null client from slipping through, but...